great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -54,6 +53,7 @@ class DataAssistant(ABC):
     def __init__(
         self,
         name: str,
+        data_assistant_result_cls: type,
```

Comment:
@NathanFarmer I think that this is an anti-pattern.  Is there any possibility to avoid this?  Happy to brainstorm together and see if anything can be done.  Thanks.

Reply:
@alexsherstinsky if a `DataAssistant` has an associated `DataAssistantResult` type, somehow it needs to know what that is. I looked into other ways and this was the most elegant I could see.

Reply:
✅ Taken care of offline